src/libs/actions/Report.js

```diff
@@ -1591,6 +1593,46 @@ function addEmojiReaction(reportID, reportActionID, emoji, skinTone = preferredS
         },
     ];
 
+    // We need to update the REPORT_ACTIONS on success
+    const originalReportID = ReportUtils.getOriginalReportID(reportID, reportAction);
+    const message = reportAction.message[0];
+
+    let reactionObject = message.reactions && _.find(message.reactions, (reaction) => reaction.emoji === emoji.name);
+    const needToInsertReactionObject = !reactionObject;
+    if (needToInsertReactionObject) {
+        reactionObject = {
+            emoji: emoji.name,
+            users: [],
+        };
+    } else {
+        // Make a copy of the reaction object so that we can modify it without mutating the original
+        reactionObject = {...reactionObject};
+    }
+
+    reactionObject.users = [...reactionObject.users, {accountID: currentUserAccountID, skinTone}];
+    let updatedReactions = [...(message.reactions || [])];
+    if (needToInsertReactionObject) {
+        updatedReactions = [...updatedReactions, reactionObject];
+    } else {
+        updatedReactions = _.map(updatedReactions, (reaction) => (reaction.emoji === emoji.name ? reactionObject : reaction));
+    }
+
+    const updatedMessage = {
+        ...message,
+        reactions: updatedReactions,
+    };
+    const successData = [
```

Comment:
Hmm why do we need to update this on success? That means that this problem will still exist when in offline mode right?

Reply:
Shouldn't we just be adding the correct optimistic data instead?

Reply:
The success data should really be coming back from the server if that is the issue

Reply:
That makes sense, @jfquevedol2198 please make the changes if you agree else let us know why, thanks!

Reply:
Done